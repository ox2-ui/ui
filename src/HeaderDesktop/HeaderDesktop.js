import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';
import Button from '@ox2/button/Button';
import FontIcon from '@ox2/icon/FontIcon';
import Thumb from '../Thumb';

/**
 * HeaderDesktop Component
 */
const HeaderDesktop = props => {
  const {
    className: classNameProp,
    onSwitchClick,
    onUserClick,
    onHomeClick,
    image,
    title,
    showSwitchMenu,
  } = props;

  const className = classNames(
    'pos:absolute-0 display:flex padding-x:7',
    classNameProp,
  );

  return (
    <div className={className}>
      <div className="flex:2 display:flex flex:items-center flex:content-start">
        <Button
          btn="large raised brand"
          className="margin-r:10"
          onClick={onHomeClick}
        >
          <FontIcon group="linear" icon="cactus" />
        </Button>
        {showSwitchMenu &&
          <div className="display:flex flex:items-center">
            <Thumb height={35} image={image} width={35} />
            <div className="size:14 text-color:white font:roboto margin-l:10">
              {title}
            </div>
            <Button
              btn="tiny neutral-dark"
              className="margin-l:10"
              onClick={onSwitchClick}
            >
              Switch
            </Button>
          </div>}
      </div>
      <div className="flex:1 display:flex flex:items-center flex:content-end">
        <Button
          btn="transparent large white"
          onClick={onUserClick}
        >
          <FontIcon group="linear" icon="user" />
        </Button>
      </div>
    </div>
  );
};

HeaderDesktop.defaultProps = {
  className: '',
  onSwitchClick: () => null,
  showSwitchMenu: false,
};

HeaderDesktop.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string.isRequired,
  onHomeClick: PropTypes.func.isRequired,
  onSwitchClick: PropTypes.func,
  onUserClick: PropTypes.func.isRequired,
  showSwitchMenu: PropTypes.bool,
  title: PropTypes.string.isRequired,
};

export default HeaderDesktop;
