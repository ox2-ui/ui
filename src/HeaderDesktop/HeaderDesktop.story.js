import React from 'react';
import { storiesOf } from '@storybook/react';
import HeaderDesktop from './HeaderDesktop';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('HeaderDesktop', module)
  .add('default', () =>
    <div style={{ height: '64px', position: 'relative' }}>
      <HeaderDesktop
        className="color:backdrop-alt"
        image={'http://imgur.com/abQ9EaG.png'}
        onHomeClick={handler}
        onSwitchClick={handler}
        onUserClick={handler}
        showSwitchMenu={true}
        title={'Some title'}
      />
    </div>,
  )
  .add('no switch', () =>
    <div style={{ height: '64px', position: 'relative' }}>
      <HeaderDesktop
        className="color:backdrop-alt"
        image={'http://imgur.com/abQ9EaG.png'}
        onHomeClick={handler}
        onUserClick={handler}
        title={'Some title'}
      />
    </div>,
  );
