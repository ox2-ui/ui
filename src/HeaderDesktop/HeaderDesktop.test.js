/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import HeaderDesktop from './HeaderDesktop';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <HeaderDesktop
      image={'http://imgur.com/abQ9EaG.png'}
      title={'Some title'}
      onHomeClick={handler}
      onSwitchClick={handler}
      onUserClick={handler}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
