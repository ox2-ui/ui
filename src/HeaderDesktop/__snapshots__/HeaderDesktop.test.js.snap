exports[`test renders correctly 1`] = `
<HeaderDesktop
  image="http://imgur.com/abQ9EaG.png"
  onHomeClick={[Function]}
  onSwitchClick={[Function]}
  onUserClick={[Function]}
  title="Some title">
  <div
    className="pos:absolute-0 display:flex padding-x:7">
    <div
      className="flex:2 display:flex flex:items-center flex:content-start">
      <Button
        btn="large raised brand"
        className="margin-r:10"
        onClick={[Function]}>
        <button
          className="margin-r:10"
          data-btn="large raised brand"
          onClick={[Function]}>
          <Icon
            icon="cactus">
            <span
              className="icon-cactus"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              } />
          </Icon>
        </button>
      </Button>
    </div>
    <div
      className="flex:1 display:flex flex:items-center flex:content-end">
      <Button
        btn="transparent large white"
        onClick={[Function]}>
        <button
          data-btn="transparent large white"
          onClick={[Function]}>
          <Icon
            icon="user">
            <span
              className="icon-user"
              style={
                Object {
                  "color": undefined,
                  "fontSize": undefined,
                }
              } />
          </Icon>
        </button>
      </Button>
    </div>
  </div>
</HeaderDesktop>
`;
